pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -220,6 +220,21 @@ func (r *ReplicaSetReconciler) createOrUpdateStatefulSet(mdb mdbv1.MongoDB) erro
 	return nil
 }
 
+func (r *ReplicaSetReconciler) createOrUpdateService(mdb mdbv1.MongoDB) error {
+	svc := buildService(mdb)
+	currentSvc := corev1.Service{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, &currentSvc); err == nil {
+		// For the particular case of the Service we need to provide the resourceVersion when updating the object.
+		// From the docs (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata)
+		// > "This value MUST be treated as opaque by clients and passed unmodified back to the server"
+		svc.ResourceVersion = currentSvc.ResourceVersion
+	} else if !errors.IsNotFound(err) {
```

Comment:
wouldn't it be better if we follow usual error handling pattern: if err!= nil return err

and in the rest of the code override the ResourceVersion unconditionally

Reply:
resourceVersion can't be update unconditionally, it needs to be fetched from an existing service. If the resource does not exist (this is a service that will be just created) then resourceVersion is not updated.

Reply:
At the moment, what's the use case for updating the service? Currently, once the service is created, it never needs to be be updated. This might change in the future, but can we simply do the creation, and never try and update.

Reply:
this code reminds me the one where we had some error not handled properly with imbricated elseif

what i undrestand is that if err is nil we override ResourceVersion and if it is not nil we return the error

please make the behaviour more explicit (I think starting by sending the error when it's not nil is the best start and it's the most common pattern)